src/autoscaling/snapshotter.js

```diff
@@ -222,13 +223,33 @@ class Snapshotter {
             });
 
             this.memorySnapshots.push(snapshot);
+            this._memoryOverloadWarning(memInfo);
         } catch (err) {
             log.exception(err, 'Snapshotter: Memory snapshot failed.');
         } finally {
             intervalCallback();
         }
     }
 
+    /**
+     * Checks for critical memory overload and logs it to the console.
+     * @ignore
+     * @param {Object} memoryInfo - Memory information
+     */
+    _memoryOverloadWarning(memoryInfo) {
+        try {
+            const criticalOverloadRation = this.maxUsedMemoryRatio * CRITICAL_OVERLOAD_RATIO;
+            const { mainProcessBytes, childProcessesBytes } = memoryInfo;
+            const usedBytes = mainProcessBytes + childProcessesBytes;
+            const isCriticalOverload = usedBytes / this.maxMemoryBytes > criticalOverloadRation;
+            if (isCriticalOverload) {
+                log.warning(`Memory is critical overloaded. Overload ratio: ${usedBytes / this.maxMemoryBytes}`);
```

Comment:
Can we show some better info than "overload ratio" ? E.g. "Used 4500 MB of 5000 MB (85%)"